iroh-rpc-client/src/status.rs

```diff
@@ -196,6 +203,30 @@ impl StatusTable {
     }
 }
 
+pub struct StatusTableIterator<'a> {
+    table: &'a StatusTable,
+    iter: usize,
+}
+
+impl Iterator for StatusTableIterator<'_> {
+    type Item = StatusRow;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let current = match self.iter {
+            0 => Some(self.table.store.to_owned()),
+            1 => Some(self.table.p2p.to_owned()),
+            2 => Some(self.table.gateway.to_owned()),
+            _ => None,
+        };
+
+        self.iter += 1;
+        if current == None {
+            self.iter = 0;
```

Comment:
is it on purpose that this iterator is infinite?

Reply:
actually it doesn't seem to be, I am confused about this logic here. what are you trying to do?

Reply:
I mean yes, but now I realize that that doesn't make sense b/c why would you re-use iterators in Rust. I'll remove this reset entirely